### PR TITLE
Show payments history on user profile page

### LINKED ISSUE #228: User-facing dues status
- List of dues payments made
 - Current dues status (active/not)
 - Active until X date (from dues package)

### PR DESCRIPTION
**Pull request by evan10s**
* Adds payment history on the User Profile page. Both dues and travel payments will appear in the list
  * As much transaction payment detail as available will be displayed
  * Added backend HTTP Resources for Travel and TravelAssignment resources to achieve this
  * $0 transactions without card details are omitted from the list (these generally come from incomplete Square transactions)
* Added the fancy centered loading component to the OAuth2 client / PAT tables

Test instructions:
* Go to https://apiary-test.robojackets.org/profile and look at your payments list
  * It would also be good to fabricate some transactions like refunds and/or impersonate a few other users to spot-check how it looks to them

TODO:
- [x] Change to order payments in descending `updated_at` order
- [x] Add min width on payment table columns to improve UI on small screens
- [x] Unit tests for payments API endpoint (dues transactions)
- [x] Unit tests incorporating travel payments
- [ ] Verify payments history for users in apiary-test
- [x] Fix https://robojackets.sentry.io/issues/3920778010

**Screenshots**
* Kitchen sink of transaction types
  * image
* Empty state
  * image
* Loading state
  * image
* Error state
  * image

Closes #228

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -103,6 +107,7 @@ parameters:
...
     - '#Parameter \#1 \$callback of method Laravel\\Nova\\Actions\\Action::canRun\(\) expects Closure\(Laravel\\Nova\\Http\\Requests\\NovaRequest, mixed\): bool, Closure\(Laravel\\Nova\\Http\\Requests\\NovaRequest, App\\Models\\[a-zA-Z]+\): bool given\.#'
     - '#Parameter \#1 \$channel of method Illuminate\\Notifications\\Messages\\SlackMessage::to\(\) expects string, string\|null given\.#'
+    - '#Parameter \#1 \$column of method Illuminate\\Database\\Eloquent\\Builder<App\\Models\\Payment>::where\(\) expects array<int\|string, mixed>\|\(Closure\(Illuminate\\Database\\Eloquent\\Builder<App\\Models\\Payment>\): void\)\|\(Closure\(Illuminate\\Database\\Eloquent\\Builder<App\\Models\\Payment>\): Illuminate\\Database\\Eloquent\\Builder<App\\Models\\Payment>\)\|Illuminate\\Database\\Query\\Expression\|string, Closure\(Illuminate\\Database\\Eloquent\\Builder\): Illuminate\\Database\\Eloquent\\Builder<Illuminate\\Database\\Eloquent\\Model> given\.#'
```
Comment (evan10s) [inline phpstan.neon:110]: It seems unusual to have to ignore this error but what it's referencing seems like a perfectly valid use of `where` to me:

https://github.com/RoboJackets/apiary/blob/8f02f6ecd3b5c42f17a6a69087bc81d260ec3dad/app/Http/Controllers/PaymentController.php#L70

Comment (kberzinch) [reply phpstan.neon:109]: That says the closure should return `void`. I feel like that would be more idiomatic, though also definitely more verbose than using arrow functions.

**Review by kberzinch — CHANGES_REQUESTED** [review]

**Review by JoshOldenburg — COMMENTED** [review]
Looks good overall. One question before approving.

File: app/Http/Resources/Payment.php [hunk 2]
```diff
@@ -29,10 +33,24 @@ public function toArray(Request $request): array
             'client_txn_id' => $this->client_txn_id,
             'server_txn_id' => $this->server_txn_id,
             'unique_id' => $this->unique_id,
-            'notes' => $this->notes,
+            'card_type' => $this->card_type,
+            'card_brand' => $this->card_brand,
+            'last_4' => $this->last_4,
+            'entry_method' => $this->entry_method,
+            'statement_description' => $this->statement_description,
+            'receipt_url' => $this->receipt_url,
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply app/Http/Resources/Payment.php:41]: All of this is already in Nova.

**Review by JoshOldenburg — APPROVED** [review]

**Review by kberzinch — APPROVED** [review]

**Pull request merged**